### PR TITLE
Use Python Version: Use built-in task support for output variables.  Add per-user scripts directory on Windows

### PR DESCRIPTION
**Pull request by brcrista**
* I didn't know there was built-in task support for output variables, so I am changing the task to use that instead of using a task parameter.
* Add the per-user scripts directory to PATH on Windows

### CODE REVIEW

File: Tasks/UsePythonVersion/task.json [hunk 1]
```diff
@@ -27,21 +27,19 @@
             "defaultValue": ">= 3.3",
             "helpMarkDown": "Version range or exact version of a Python version to use."
         },
```
Comment (jpricketMSFT) [inline Tasks/UsePythonVersion/task.json]: I would recommend incrementing the Patch version even though this version hasn't been released yet.

File: Tasks/UsePythonVersion/taskutil.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    Windows,
+    MacOS,
+    Linux
+}
+
+/**
+ * Determine the operating system the build agent is running on.
+ * TODO move this to vsts-task-lib
+ */
+export function getPlatform(): Platform {
```
Comment (jpricketMSFT) [inline Tasks/UsePythonVersion/taskutil.ts]: Most other tasks that have to check platform do a simple string comparison. If you do end up adding this to the task lib, we should also add a user story to clean up all the other tasks.

Comment (brcrista) [reply Tasks/UsePythonVersion/taskutil.ts]: That would be nice.  The current way as I recall is to check `taskLib.osType().match(/^Win/)` something.  But that only tells you whether you're on Windows or not Windows, and is a bit more complicated than necessary.

File: Tasks/UsePythonVersion/task.json (hunk 1 above)
Comment (brcrista) [reply Tasks/UsePythonVersion/task.json]: done

**Pull request merged**